analyzer/api.go

```diff
@@ -53,6 +55,9 @@ type BlockRange struct {
 // RuntimeConfig specifies configuration parameters for
 // processing the runtime layer.
 type RuntimeConfig struct {
```

Comment:
Since `RuntimeConfig` now aims to contain all metadata about the runtime, can you please also add `runtimeName common.Runtime` and remove it from analyzers' `Main` structs? Then for consistent naming, `ParaTime` can maybe become `runtimeInfo` or `runtimeSdkInfo`. Your call on the exact names, but let's definitely not have `runtime` for the name and `ParaTime` for other metadata :)

Background: The existence of this type (and even more so `ConsensusConfig`, seeing as how there's only one consensus) has annoyed me for a long time because it seemed to contain such a random subset of an analyzer's config. It felt to me that its fields should just be absorbed into the containing struct, which is always an analyzer `Main`. But well ... after all these cleanups that we've done, and after introducing all these new analyzers that deal with the runtime(s), the separation seems to make good sense. All this struct is missing is the runtime name.

Reply:
ok let's try that. might be a little awkward fit for the token and token balance analyzers, which don't use this yet. they don't use the block range part, but everything else I think is in common

Reply:
thusly moved